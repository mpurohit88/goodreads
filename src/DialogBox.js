import React, {Fragment} from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

class DialogBox extends React.Component {
  state = {
    open: true,
    scroll: 'paper',
  };

  handleClickOpen = scroll => () => {
    this.setState({ open: true, scroll });
  };

  handleClose = () => {
    this.setState({ open: false });
    this.props.reset();
  };

  render() {
    const { bookDetail } = this.props;

    return (
      <div>
        <Dialog
          open={this.state.open}
          onClose={this.handleClose}
          scroll={this.state.scroll}
          aria-labelledby="scroll-dialog-title"
        > 
        {
          bookDetail ? 
          <span> 
            <DialogTitle id="scroll-dialog-title">Bood Detail</DialogTitle>
            <DialogContent>
              <Grid container spacing={16}>
                <Grid item>
                    <img  alt="complex" src={bookDetail.image_url} />
                </Grid>
                <Grid item xs={12} sm container>
                  <Grid item xs container direction="column" spacing={16}>
                    <Grid item xs>
                      <Typography gutterBottom variant="h5">
                        {bookDetail.title['#cdata']}
                      </Typography>
                      <Typography gutterBottom variant="h5">
                        <b>by {Array.isArray(bookDetail.authors.author) ? bookDetail.authors.author.map(obj => {return obj.name + ", ";}) : bookDetail.authors.author.name} </b>
                      </Typography>
                      <Typography color="textSecondary">*****: {bookDetail.average_rating} | Rating details | {bookDetail.ratings_count["#cdata"]} Ratings | {bookDetail.text_reviews_count["#cdata"]} Reviews</Typography>
                    </Grid>
                  </Grid>
                </Grid>
              </Grid>
              <br />
              <Fragment><span dangerouslySetInnerHTML={{__html: bookDetail.description && bookDetail.description['#cdata']}} /></Fragment>
            </DialogContent>
            <DialogActions>
              <Button onClick={this.handleClose} color="primary">
                Close
              </Button>
            </DialogActions>
            </span>
          :
          <span>Loading...</span>
        }
        </Dialog>
      </div>
    );
  }
}

export default DialogBox;
