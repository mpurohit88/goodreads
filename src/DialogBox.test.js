import React from 'react';
import ReactDOM from 'react-dom';
import DialogBox from './DialogBox';
import { shallow } from 'enzyme';
import { } from './setupTests';

it('renders without crashing when bookDetail is undefined', () => {
    const bookDetail = undefined;
    const div = document.createElement('div');
    ReactDOM.render(<DialogBox bookDetail={bookDetail} reset={() => {}}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

it('render DialogBox correctly', () => {
    const wrapper = shallow(<DialogBox />);
    expect(wrapper).toMatchSnapshot();
  });

it('renders without crashing', () => {
  const bookDetail = {title: {"#cdata": "book title"}, text_reviews_count: {"#text": "324"}, ratings_count: {"#text": "23"}, average_rating: 123,
                        best_book: {image_url: 'https://www.goodreads.com/img.png'}, authors: {author: {name: "React"}}}
  const div = document.createElement('div');
  ReactDOM.render(<DialogBox bookDetail={bookDetail} reset={() => {}}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});