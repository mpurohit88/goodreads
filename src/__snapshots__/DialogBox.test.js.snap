// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DialogBox correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DialogBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-labelledby="scroll-dialog-title"
        onClose={[Function]}
        open={true}
        scroll="paper"
      >
        <span>
          Loading...
        </span>
      </WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "scroll-dialog-title",
        "children": <span>
          Loading...
        </span>,
        "onClose": [Function],
        "open": true,
        "scroll": "paper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "span",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-labelledby="scroll-dialog-title"
          onClose={[Function]}
          open={true}
          scroll="paper"
        >
          <span>
            Loading...
          </span>
        </WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "scroll-dialog-title",
          "children": <span>
            Loading...
          </span>,
          "onClose": [Function],
          "open": true,
          "scroll": "paper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "span",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
